prog/dftb+/lib_dftb/scc.F90

```diff
@@ -1332,9 +1332,11 @@ subroutine initGamma_(this, env, species, iNeighbour)
     ! some additional symmetry not used in these loops, as the value of gamma for atoms interacting
     ! with themselves is the same for all atoms of the same species
 
+    ! Workaround for bug in Intel18 compiler can not use OMP REDUCTION(:+work) in debug mode
```

Comment:
This is not a bug in Intel18, I don't see a + reduction on `work` in your implementation, it _should_ be shared.

Reply:
See the previous commit [6faf7d36](https://github.com/bhourahine/dftbplus/commit/6faf7d36c459b639946dfc97ded52c58dfa276ab)
which works for intel19 but not 18, weirdly showing up as a segfault on a test using GPUs.

Reply:
Weird segfault is usually an omp stacksize problem. The GPU testcases are huge and an omp reduction can be quite wasteful with memory, especially when it is not needed in this context. We could try it again and increase the `OMP_STACKSIZE` environment variable, which is usually set by default to a (too) small value. Point is, that is not a bug, but some weird default behaviour of the compiler (which tend to change between releases, unfortunately).

Reply:
The matrix being reduced is quite small (setting the OMP stack size to 16M is enough to avoid the segfault). Removing the reduction operation runs a little bit slower, at least on two threads (due to a difference in cache behaviour presumably, since each thread isn't needing to write against the same matrix until the reduce at the end) but avoids that segfault.

Reply:
If you ever find a way to set (or even check) the stacksize from Fortran, let me know.